internal/exec/template_funcs_component.go

```diff
@@ -4,32 +4,39 @@ import (
 	"fmt"
 	"sync"
 
+	log "github.com/charmbracelet/log"
 	"github.com/samber/lo"
 
+	cfg "github.com/cloudposse/atmos/pkg/config"
 	"github.com/cloudposse/atmos/pkg/schema"
 	u "github.com/cloudposse/atmos/pkg/utils"
 )
 
 var componentFuncSyncMap = sync.Map{}
 
-func componentFunc(atmosConfig schema.AtmosConfiguration, component string, stack string) (any, error) {
-	u.LogTrace(fmt.Sprintf("Executing template function 'atmos.Component(%s, %s)'", component, stack))
+func componentFunc(
+	atmosConfig *schema.AtmosConfiguration,
+	configAndStacksInfo *schema.ConfigAndStacksInfo,
+	component string,
+	stack string,
+) (any, error) {
+	log.Debug(fmt.Sprintf("Executing template function 'atmos.Component(%s, %s)'", component, stack))
 
 	stackSlug := fmt.Sprintf("%s-%s", stack, component)
 
 	// If the result for the component in the stack already exists in the cache, return it
 	existingSections, found := componentFuncSyncMap.Load(stackSlug)
 	if found && existingSections != nil {
-		if atmosConfig.Logs.Level == u.LogLevelTrace {
-			u.LogTrace(fmt.Sprintf("Found the result of the template function 'atmos.Component(%s, %s)' in the cache", component, stack))
+		if atmosConfig.Logs.Level == u.LogLevelDebug {
+			log.Debug(fmt.Sprintf("Found the result of the template function 'atmos.Component(%s, %s)' in the cache", component, stack))
```

Comment:
Use structured logging

Reply:
Charmbracelet log supports multi-line events.

Reply:
I would need to see an example of why we shouldn't use the native built in multi line handling.